server/src/utils/userUtils.js

```diff
@@ -50,10 +50,41 @@ const removeUserById = (
   removeUserFromRoom(user, rooms, io); // handles transfer owner
 }
 
+const getUserId = (
+  username: string,
+  roomId: string,
+  users: UserSetT
+): ?string => {
```

Comment:
I don't like to iterate through all the users to find the owner. As I understand, we will search for the owner of the room when a user joins a room, right? If we have a lot of users this will be slower and slower. But, without changing something we do not have another option. 

Maybe we can store the user object for the owner or the owner id inside the room object, or have an aux dictionary with the room id as key, and the owner object as value, this way we do not need to change things for the front end, but it will be a little messy. 

We can work with this for now, but let's think about it.

Reply:
I agree.
But it might be trickier than that,
Lets suppose we have the owner user object inside the room.
When we transfer ownership we'll have to make the same search for the new owner, because we only have their username.
Although, transfer ownership is less frequent than user joined.
I think the root problem is that we don't have a reverse look up table for usernames.

I also agree this is ok for now, future refactoring will be needed